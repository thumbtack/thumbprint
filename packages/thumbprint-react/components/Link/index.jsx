import React from 'react';
import PropTypes from 'prop-types';
import warning from 'warning';
import { Themed, Plain } from '../UIAction/index';

const getCommonLinkProps = props => {
    warning(
        // If `onClick` prop exists then `to` must also exist.
        !props.onClick || (props.to && props.onClick),
        'The `onClick` prop in the Thumbprint `Link` and `ThemedLink` components should only be used when the `to` prop is also provided. Consider using the `Button` component instead.',
    );

    return {
        to: props.to,
        onClick: props.onClick,
        shouldOpenInNewTab: props.shouldOpenInNewTab,
        isDisabled: props.isDisabled,
        children: props.children,
        accessibilityLabel: props.accessibilityLabel,
        dataTest: props.dataTest,
    };
};

/**
 * Anchor link that renders as text.
 */
const Link = React.forwardRef((props, ref) => (
    <Plain
        {...getCommonLinkProps(props)}
        theme={props.theme}
        iconLeft={props.iconLeft}
        iconRight={props.iconRight}
        ref={ref}
    />
));

Link.propTypes = {
    /**
     * Contents displayed within the anchor.
     */
    children: PropTypes.node,
    /**
     * Description of the link’s content. It is required if the link contains an icon and no
     * descriptive text.
     */
    accessibilityLabel: PropTypes.string,
    /**
     * Functionally disables the anchor. We discourage the use of this prop since it is difficult
     * to visually indicate that a link is disabled. Consider not rendering the link if it is not
     * interactive.
     */
    isDisabled: PropTypes.bool,
    /**
     * Page to navigate to when the anchor is clicked.
     */
    to: PropTypes.string,
    /**
     * Function to fire when clicking on the anchor. This should be used alongside the `to` prop.
     */
    onClick: PropTypes.func,
    /**
     * Icon from [Thumbprint Icons](/icons/) to render left of the text within `Link`.
     */
    iconLeft: PropTypes.node,
    /**
     * Icon from [Thumbprint Icons](/icons/) to render right of the text within `Link`.
     */
    iconRight: PropTypes.node,
    /**
     * Opens the URL in a new tab when clicked.
     */
    shouldOpenInNewTab: PropTypes.bool,
    /**
     * Sets the anchor’s text color.
     *
     * `inherit` will make the anchor inherit `color` from its parent.
     */
    theme: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'inherit']),
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest: PropTypes.string,
};

Link.defaultProps = {
    children: undefined,
    accessibilityLabel: undefined,
    isDisabled: false,
    iconLeft: null,
    iconRight: null,
    to: null,
    onClick: undefined,
    theme: 'primary',
    shouldOpenInNewTab: false,
    dataTest: undefined,
};

/**
 * Anchor link that visually looks like a button.
 */
const ThemedLink = React.forwardRef((props, ref) => (
    <Themed
        {...getCommonLinkProps(props)}
        icon={props.icon}
        theme={props.theme}
        size={props.size}
        width={props.width}
        ref={ref}
    />
));

ThemedLink.propTypes = {
    /**
     * Contents displayed within the button.
     */
    children: PropTypes.node,
    /**
     * Description of the link’s content. It is required if the link contains an icon and no
     * descriptive text.
     */
    accessibilityLabel: PropTypes.string,
    /**
     * Visually and functionally disables the button.
     */
    isDisabled: PropTypes.bool,
    /**
     * Page to navigate to when the anchor is clicked.
     */
    to: PropTypes.string,
    /**
     * Function to fire when clicking on the anchor. This should be used alongside the `to` prop.
     */
    onClick: PropTypes.func,
    /**
     * Opens the URL in a new tab when clicked.
     */
    shouldOpenInNewTab: PropTypes.bool,
    /**
     * Icon from [Thumbprint Icons](/icons/) to render within the
     * button. It must be one of the `small` icons.
     */
    icon: PropTypes.node,
    /**
     * Controls the button's background, text, and border theme.
     */
    theme: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'caution', 'solid']),
    /**
     * Changes the button's `line-height`, `padding`, `border-radius`, and `font-size`.
     */
    size: PropTypes.oneOf(['small', 'large']),
    /**
     * Themed links are as wide as the content that is passed in.  The `full` option will
     * expand the width to `100%` on all screens. `full-below-small` will expand the width to 100%
     * on devices smaller than [our `small` breakpoint](/tokens/#section-breakpoint).
     */
    width: PropTypes.oneOf(['auto', 'full', 'full-below-small']),
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest: PropTypes.string,
};

ThemedLink.defaultProps = {
    children: undefined,
    accessibilityLabel: undefined,
    isDisabled: false,
    shouldOpenInNewTab: false,
    theme: 'primary',
    size: 'large',
    width: 'auto',
    icon: null,
    to: null,
    onClick: undefined,
    dataTest: undefined,
};

export default Link;
export { ThemedLink };
