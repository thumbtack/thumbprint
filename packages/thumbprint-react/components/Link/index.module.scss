@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';

.primary {
    color: $tp-color__black;
    text-decoration: underline;
}

.secondary {
    color: $tp-color__black;
}
