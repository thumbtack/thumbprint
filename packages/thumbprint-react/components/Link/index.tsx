import React from 'react';
import warning from 'warning';
import { Themed, Plain } from '../UIAction/index';

interface CommonProps {
    to?: string;
    target?: string;
    shouldOpenInNewTab?: boolean;
    children?: React.ReactNode;
    isDisabled?: boolean;
    onClick?: () => void;
    dataTest?: string;
    accessibilityLabel?: string;
}

const getCommonLinkProps = (props: CommonProps): CommonProps => {
    warning(
        // If `onClick` prop exists then `to` must also exist.
        !props.onClick || (props.to && props.onClick),
        'The `onClick` prop in the Thumbprint `Link` and `ThemedLink` components should only be used when the `to` prop is also provided. Consider using the `Button` component instead.',
    );

    return {
        to: props.to,
        onClick: props.onClick,
        target: props.target,
        shouldOpenInNewTab: props.shouldOpenInNewTab,
        isDisabled: props.isDisabled,
        children: props.children,
        accessibilityLabel: props.accessibilityLabel,
        dataTest: props.dataTest,
    };
};

/**
 * Anchor link that renders as text.
 */
const Link = React.forwardRef<HTMLAnchorElement, LinkPropTypes>(
    (
        {
            to,
            onClick,
            target,
            shouldOpenInNewTab = false,
            isDisabled = false,
            children,
            accessibilityLabel,
            dataTest,
            theme = 'primary',
            iconLeft,
            iconRight,
        }: LinkPropTypes,
        ref,
    ) => (
        <Plain
            {...getCommonLinkProps({
                to,
                onClick,
                target,
                shouldOpenInNewTab,
                isDisabled,
                children,
                accessibilityLabel,
                dataTest,
            })}
            theme={theme}
            iconLeft={iconLeft}
            iconRight={iconRight}
            ref={ref}
        />
    ),
);

interface LinkPropTypes {
    /**
     * Contents displayed within the anchor.
     */
    children?: React.ReactNode;
    /**
     * Description of the link’s content. It is required if the link contains an icon and no
     * descriptive text.
     */
    accessibilityLabel?: string;
    /**
     * Functionally disables the anchor. We discourage the use of this prop since it is difficult
     * to visually indicate that a link is disabled. Consider not rendering the link if it is not
     * interactive.
     */
    isDisabled?: boolean;
    /**
     * Page to navigate to when the anchor is clicked.
     */
    to?: string;
    /**
     * Function to fire when clicking on the anchor. This should be used alongside the `to` prop.
     */
    onClick?: () => void;
    /**
     * Icon from [Thumbprint Icons](/icons/) to render left of the text within `Link`.
     */
    iconLeft?: React.ReactNode;
    /**
     * Icon from [Thumbprint Icons](/icons/) to render right of the text within `Link`.
     */
    iconRight?: React.ReactNode;
    /**
     * The anchor `target` attribute. Set this to `_blank` to open in a new tab, or to an arbitrary
     * string to open the link in an `<iframe>` with the same `name`.
     */
    target?: string;
    /**
     * @deprecated
     * Opens the URL in a new tab when clicked.
     */
    shouldOpenInNewTab?: boolean;
    /**
     * Sets the anchor’s text color.
     *
     * `inherit` will make the anchor inherit `color` from its parent.
     */
    theme?: 'primary' | 'secondary' | 'tertiary' | 'inherit';
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest?: string;
}

/**
 * Anchor link that visually looks like a button.
 */
const ThemedLink = React.forwardRef<HTMLAnchorElement, ThemedLinkPropTypes>(
    (
        {
            to,
            onClick,
            shouldOpenInNewTab = false,
            target,
            isDisabled = false,
            children,
            accessibilityLabel,
            dataTest,
            icon,
            iconRight,
            theme = 'primary',
            size = 'large',
            width = 'auto',
        }: ThemedLinkPropTypes,
        ref,
    ) => (
        <Themed
            {...getCommonLinkProps({
                to,
                onClick,
                shouldOpenInNewTab,
                target,
                isDisabled,
                children,
                accessibilityLabel,
                dataTest,
            })}
            icon={icon}
            iconRight={iconRight}
            theme={theme}
            size={size}
            width={width}
            ref={ref}
        />
    ),
);

interface ThemedLinkPropTypes {
    /**
     * Contents displayed within the button.
     */
    children?: React.ReactNode;
    /**
     * Description of the link’s content. It is required if the link contains an icon and no
     * descriptive text.
     */
    accessibilityLabel?: string;
    /**
     * Visually and functionally disables the button.
     */
    isDisabled?: boolean;
    /**
     * Page to navigate to when the anchor is clicked.
     */
    to?: string;
    /**
     * Function to fire when clicking on the anchor. This should be used alongside the `to` prop.
     */
    onClick?: () => void;
    /**
     * The anchor `target` attribute. Set this to `_blank` to open in a new tab, or to an arbitrary
     * string to open the link in an `<iframe>` with the same `name`.
     */
    target?: string;
    /**
     * Opens the URL in a new tab when clicked.
     */
    shouldOpenInNewTab?: boolean;
    /**
     * Icon from [Thumbprint Icons](/icons/) to render left within the
     * button. It must be one of the `small` icons.
     */
    icon?: React.ReactNode;
    /**
     * Icon from [Thumbprint Icons](/icons/) to render right within the
     * button. It must be one of the `small` icons.
     */
    iconRight?: React.ReactNode;
    /**
     * Controls the button's background, text, and border theme.
     */
    theme?: 'primary' | 'secondary' | 'tertiary' | 'caution' | 'solid';
    /**
     * Changes the button's `line-height`, `padding`, `border-radius`, and `font-size`.
     */
    size?: 'small' | 'large';
    /**
     * Themed links are as wide as the content that is passed in.  The `full` option will
     * expand the width to `100%` on all screens. `full-below-small` will expand the width to 100%
     * on devices smaller than [our `small` breakpoint](/tokens/#section-breakpoint).
     */
    width?: 'auto' | 'full' | 'full-below-small';
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest?: string;
}

export default Link;
export { ThemedLink };
