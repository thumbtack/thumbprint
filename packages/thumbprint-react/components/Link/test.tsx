import React from 'react';
import { mount } from 'enzyme';
import Link, { ThemedLink } from './index';

test('renders `children` passed in', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapper = mount(<Component to="https://example.com/">Goose</Component>);
        expect(wrapper.text()).toBe('Goose');
        expect(wrapper).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('renders a anchor tag when a `to` is provided', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapper = mount(<Component to="https://example.com/">Goose</Component>);
        expect(wrapper.find('button')).toHaveLength(0);
        expect(wrapper.find('a')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('renders an anchor tag when both `to` and `onClick` are provided', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapper = mount(
            <Component to="https://example.com/" onClick={jest.fn}>
                Goose
            </Component>,
        );
        expect(wrapper.find('button')).toHaveLength(0);
        expect(wrapper.find('a')).toHaveLength(1);
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('adds `rel` attribute to handle security vulnerability for `shouldOpenInNewTab`', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component shouldOpenInNewTab to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('rel')).toEqual('noopener noreferrer');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('adds `rel` attribute to handle security vulnerability for `target=_blank`', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component target="_blank" to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('rel')).toEqual('noopener noreferrer');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('adds attribute to open link in new tab', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component shouldOpenInNewTab to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('target')).toEqual('_blank');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('sets `target` attribute', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component target="foo" to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('target')).toEqual('foo');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('prefers `target` attribute to `shouldOpenInNewTab` when both are provided', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component shouldOpenInNewTab target="foo" to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('target')).toEqual('foo');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('renders `to` prop as `href`', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(<Component to="https://example.com/">Goose</Component>);
        expect(wrapperLink.find('a').prop('href')).toBe('https://example.com/');
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('renders `onClick` prop', () => {
    const testComponent = (Component: React.ElementType): void => {
        const onClick = jest.fn();
        const wrapper = mount(
            <Component to="https://example.com/" onClick={onClick}>
                Goose
            </Component>,
        );
        expect(onClick).toHaveBeenCalledTimes(0);
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('removes `href` if link is disabled', () => {
    const testComponent = (Component: React.ElementType): void => {
        const wrapperLink = mount(
            <Component isDisabled to="https://example.com/">
                Goose
            </Component>,
        );
        expect(wrapperLink.find('a').prop('href')).toBeFalsy();
        expect(wrapperLink).toMatchSnapshot();
    };

    testComponent(Link);
    testComponent(ThemedLink);
});

test('adds `dataTest` prop', () => {
    const wrapperA = mount(
        <Link to="https://example.com/" dataTest="Duck">
            Goose
        </Link>,
    );
    expect(wrapperA.find('a').prop('data-test')).toEqual('Duck');
    expect(wrapperA).toMatchSnapshot();

    const wrapperB = mount(
        <ThemedLink to="#" dataTest="Duck">
            Goose
        </ThemedLink>,
    );
    expect(wrapperB.find('a').prop('data-test')).toEqual('Duck');
    expect(wrapperB).toMatchSnapshot();
});
