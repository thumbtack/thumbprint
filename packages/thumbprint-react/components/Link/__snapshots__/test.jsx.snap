// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds \`dataTest\` prop 1`] = `
<ForwardRef
  dataTest="Duck"
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    dataTest="Duck"
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      data-test="Duck"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`adds \`dataTest\` prop 2`] = `
<ForwardRef
  dataTest="Duck"
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  size="large"
  theme="primary"
  to="#"
  width="auto"
>
  <ForwardRef
    dataTest="Duck"
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    size="large"
    theme="primary"
    to="#"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      data-test="Duck"
      disabled={false}
      href="#"
      target="_self"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`adds \`rel\` attribute to handle security vulnerability 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={true}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={true}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={false}
      href="https://example.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`adds \`rel\` attribute to handle security vulnerability 2`] = `
<ForwardRef
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={true}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={true}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={false}
      href="https://example.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`adds attribute to open link in new tab 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={true}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={true}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={false}
      href="https://example.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`adds attribute to open link in new tab 2`] = `
<ForwardRef
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={true}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={true}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={false}
      href="https://example.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`removes \`href\` if link is disabled 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={true}
  shouldOpenInNewTab={false}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={true}
    shouldOpenInNewTab={false}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={true}
      target="_self"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`removes \`href\` if link is disabled 2`] = `
<ForwardRef
  icon={null}
  isDisabled={true}
  shouldOpenInNewTab={false}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={true}
    shouldOpenInNewTab={false}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={true}
      target="_self"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders \`children\` passed in 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders \`children\` passed in 2`] = `
<ForwardRef
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders \`to\` prop as \`href\` 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders \`to\` prop as \`href\` 2`] = `
<ForwardRef
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders a anchor tag when a \`to\` is provided 1`] = `
<ForwardRef
  iconLeft={null}
  iconRight={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  theme="primary"
  to="https://example.com/"
>
  <ForwardRef
    iconLeft={null}
    iconRight={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    theme="primary"
    to="https://example.com/"
  >
    <a
      className="plain plainThemePrimary"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      Goose
    </a>
  </ForwardRef>
</ForwardRef>
`;

exports[`renders a anchor tag when a \`to\` is provided 2`] = `
<ForwardRef
  icon={null}
  isDisabled={false}
  shouldOpenInNewTab={false}
  size="large"
  theme="primary"
  to="https://example.com/"
  width="auto"
>
  <ForwardRef
    icon={null}
    isDisabled={false}
    shouldOpenInNewTab={false}
    size="large"
    theme="primary"
    to="https://example.com/"
    width="auto"
  >
    <a
      className="themedButton themedButtonRoundedBordersLeft themedButtonRoundedBordersRight themedButtonThemePrimary themedButtonWidthAuto"
      disabled={false}
      href="https://example.com/"
      target="_self"
    >
      <span
        className="flexWrapper flexWrapperSizeLarge"
      >
        Goose
      </span>
    </a>
  </ForwardRef>
</ForwardRef>
`;
